src/lib/ngc/compile-source-files.ts

```diff
@@ -69,13 +71,24 @@ export async function compileSourceFiles(
 
   cache.oldPrograms = { ...cache.oldPrograms, [scriptTarget]: ngProgram };
 
-  const exitCode = ng.exitCodeFromResult(diagnostics);
-  if (exitCode !== 0) {
-    throw new Error(ng.formatDiagnostics(diagnostics));
+  if (!hasError) {
+    const { diagnostics } = ngProgram.emit({
+      emitCallback: createEmitCallback(tsConfigOptions),
+      emitFlags
+    });
+
+    allDiagnostics.push(...diagnostics);
   }
 
-  ngProgram.emit({
-    emitCallback: createEmitCallback(tsConfigOptions),
-    emitFlags
-  });
+  if (allDiagnostics.length === 0) {
+    return;
+  }
+
+  const exitCode = ng.exitCodeFromResult(allDiagnostics);
```

Comment:
I was misled in thinking that this is the same result as on line 64, however new diagnostics may have been added during emit (the whole point of this PR). Can we add some comments to guide the reader (or just me perhaps 🤷‍♂️) what is going on here :)

Reply:
Yeah I’ll add them 😊